docs/src/extend/plugins.md

```diff
@@ -8,180 +8,249 @@ eleventyNavigation:
+The `meta.name` property should match the npm package name for your plugin and the `meta.version` property should match the npm package version for your plugin. The easiest way to accomplish this is by reading this information from your `package.json`.
 
-### Environments in Plugins
+::: tip
+While there are no restrictions on plugin names, it helps others to find your plugin on [npm](https://npmjs.com) when you follow these naming conventions:
 
-Plugins can expose additional environments for use in ESLint. To do so, the plugin must export an `environments` object. The keys of the `environments` object are the names of the different environments provided and the values are the environment settings. For example:
+* **Unscoped:** If your npm package name won't be scoped (doesn't begin with `@`), then the plugin name should begin with `eslint-plugin-`, such as `eslint-plugin-example`.
+* **Scoped:** If your npm package name will be scoped, then the plugin name should be in the format of `@<scope>/eslint-plugin-<plugin-name>` such as `@jquery/eslint-plugin-jquery` or even `@<scope>/eslint-plugin` such as `@jquery/eslint-plugin`.
+:::
+
+As an alternative, you can also expose `name` and `version` properties at the root of your plugin, such as:
 
 ```js
-module.exports = {
-    environments: {
-        jquery: {
-            globals: {
-                $: false
-            }
-        }
+const plugin = {
+
+    // alternate location of name and version
+    name: "eslint-plugin-example",
+    version: "1.2.3"
+    rules: {
+        // add rules here
     }
 };
+
+// for ESM
+export default plugin;
+
+// OR for CommonJS
+module.exports = plugin;
 ```
 
-There's a `jquery` environment defined in this plugin. To use the environment in ESLint, you would use the unprefixed plugin name, followed by a slash, followed by the environment name. So if this plugin were named `eslint-plugin-myplugin`, then you would set the environment in your configuration to be `"myplugin/jquery"`.
+::: important
+While the `meta` object is the preferred way to provide the plugin name and version, this format is also acceptable and is provided for backward compatibility.
+:::
 
-Plugin environments can define the following objects:
+### Rules in Plugins
 
-1. `globals` - acts the same `globals` in a configuration file. The keys are the names of the globals and the values are `true` to allow the global to be overwritten and `false` to disallow.
-1. `parserOptions` - acts the same as `parserOptions` in a configuration file.
+Plugins can expose custom rules for use in ESLint. To do so, the plugin must export a `rules` object containing a key-value mapping of rule ID to rule. The rule ID does not have to follow any naming convention (so it can just be `dollar-sign`, for instance). To learn more about creating custom rules in plugins, refer to [Custom Rules](custom-rules).
```

Comment:
Perhaps we should mention that the use of `/` in rule ID is not supported because https://github.com/eslint/eslint/discussions/17901#discussioncomment-7930321

Reply:
Looks like parsing is different when the plugin doesn't start with `@` (i.e., the plugin is not scoped), so `/` in rule ID would work in that case.

https://github.com/eslint/eslint/blob/3826cdf89294d079be037a9ab30b7506077b26ac/lib/config/flat-config-helpers.js#L24-L29

```js
// this works
module.exports = {
    plugins: {
        "foo": {
            rules: {
                "bar/baz": {
                    create() {
                        return {}
                    }
                }
            }
        }
    },
    rules: {
        "foo/bar/baz": 2
    }
};

// this throws TypeError: Key "rules": Key "@foo/bar/baz": Could not find plugin "@foo/bar"
module.exports = {
    plugins: {
        "@foo": {
            rules: {
                "bar/baz": {
                    create() {
                        return {}
                    }
                }
            }
        }
    },
    rules: {
        "@foo/bar/baz": 2
    }
};
```

Nevertheless, it does seem best to generally say that rule IDs should not contain `/`, to avoid ambiguity.

Reply:
Yeah, I need to do this to mimic the behavior of eslintrc plugin names. I've added a note about the slash.

Reply:
Also added a note about the `@` restrictions in namespaces.